### PR TITLE
Dictionary learning

### PR DESCRIPTION
**Pull request by TuKo**
This PR implements the work from INRIA on Dictionary learning for atlas parcellation.
Example is not included as it is planned to be added later when Cameron's paper is ready.

### CODE REVIEW

**Comment by mihaic** [comment]
@cameronphchen, @hejiaz, you should be familiar with this. Could you please review?

File: brainiak/funcalign/dl.py [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+The implementation is based on the following publication:
+
+.. [Varoquaux2011] "Multi-subject Dictionary Learning to Segment an Atlas of
+   Brain Spontaneous Activity",
+   G. Varoquaux, A. Gramfort, F. Pedregosa, V. Michel, .B. Thirion
+   22nd International Conference Information Processing in Medical Imaging
+   (IPMI 2011)
+   http://rd.springer.com/chapter/10.1007/978-3-642-22092-0_46
+
+"""
```
Comment (mihaic) [inline brainiak/funcalign/dl.py]: Please add a paragraph stating this code implements a different algorithm from nilearn.

**Comment by cameronphchen** [comment]
@mihaic @TuKo , sure! I can do the review.

**Comment by mihaic** [comment]
@cameronphchen, please remember this PR.

**Comment by TuKo** [comment]
@cameronphchen 
I checked for convergence of the algorithm with raider in an unusual way.
This method requires the position of the voxels in the brain, compared to SRM that doesn't. 

So unfortunately I don't have data to test for significance results of this method and I'm expecting to add an example further on, when working on the survey paper.

**Comment by cameronphchen** [comment]
@TuKo @mihaic  If that's the case, I don't think we should merge this into the master branch at this point since we can't guarantee its correctness. In other words, we haven't had a success story of using this code. I can work on the example later on, but it's probably better we merge this with the example together. What do you think?

**Comment by TuKo** [comment]
Ok, we will wait for your example to accept the PR.

**Comment by mihaic** [comment]
@cameronphchen, how is the example coming along?

**Comment by cameronphchen** [comment]
@mihaic:  Hejia @hejiaz  has tested the code but there seem to be some issues with the code leading to incorrect results. We have discussed this with Javier @TuKo.

**Comment by mihaic** [comment]
@cameronphchen, @hejiaz, thanks for your help. Javier and I decided to close this PR because the algorithm it implements has been superseded by better algorithms, so it would not be of interest to BrainIAK users. Feel free to reopen if you disagree.

**Comment by cameronphchen** [comment]
Sounds good to me. Thanks!

**Pull request closed without merging**